### ISSUE TITLE
Barbarians incorrect countdown, invisible - citizens remain in bed.

### ISSUE THREAD

**Issue by QueenBonnydust** [ldtteam/minecolonies#2774]
### Minecolonies version

minecolonies-1.12.2-0.9.86-ALPHA-universal.jar 

### Expected behavior

- Run raid-tonight command for the colony or allow a raid to occur naturally.
- Kill barbarians and see countdown of remaining barbarians in chat followed by successfully defended your colony message.
- Citizens return to work.

### Actual behaviour

- Run raid-tonight command for the colony or allow a raid to occur naturally.
- Kill barbarians, countdown of barbarians in chat does not equal amount of actual barbarians.
- Citizens do not return to work.

Example - Run raid tonight command. 3 barbarians visible, kill one, kill second one, message in chat comes up 3 barbarians remaining, kill third one, message in chat says 2 barbarians remaining, citizens do not return to work and are still in beds or frozen positions. Refresh colony has no effect (not sure if its appropriate for this use anyway), citizens recalled to town hall immediately return to beds or frozen positions. Allow night to pass by sleeping. Sleep through another night or two with no change.

Run command again. 4 barbarians visible, kill one, chat message five barbarians remaining, kill second one, chat message four barbarians remaining, kill third one, chat message three barbarians remaining, kill fourth one, chat message two barbarians remaining, citizens do not return to work and are still in beds or frozen positions.

Restart server, citizens return to work.

Console during this example:

```
>     [33;1m[11:10:41] [Server thread/WARN] [minecolonies]: Spawning barbarian at: BlockPos{x=1133, y=63, z=-720}
[39;0m>     [33;1m[11:10:41] [Server thread/WARN] [minecolonies]: Spawning barbarian at: BlockPos{x=1133, y=63, z=-720}
[39;0m>
[39;0m>     [33;1m[11:12:12] [Server thread/WARN] [minecolonies]: Spawning barbarian at: BlockPos{x=1187, y=62, z=-711}
[39;0m>
[39;0m>     [33;1m[11:12:31] [Server thread/WARN] [minecolonies]: Spawning barbarian at: BlockPos{x=1190, y=62, z=-713}
[39;0m>
[39;0m>
>
>
>
>
>
>
>
>
>
>
>
>
>
>
>
>
>
>
>
>
>
>
>
>
>
>
>
>
>
>
>
>
>
>
>
>
>
>     [11:39:26] [Server thread/INFO] [minecolonies]: [BarbarianEvent]: Spawning: 320 1476
>
>
>
>
>
>
>
>
>
>
>
>
>
>
>
>
>
>     [33;1m[11:50:20] [Server thread/WARN] [minecolonies]: Spawning barbarian at: BlockPos{x=1359, y=63, z=-721}
[39;0m>     [33;1m[11:50:20] [Server thread/WARN] [minecolonies]: Spawning barbarian at: BlockPos{x=1359, y=63, z=-721}
[39;0m>
>
>
>
>
>
Console > 

Log from this example:

[10:59:49] [Server thread/INFO] [FML]: [Server thread] Server side modded connection established
[10:59:49] [Server thread/INFO] [net.minecraft.server.management.PlayerList]: bonnydust[/49.193.30.198:57740] logged in with entity id 123251 at (364.30000001192093, 73.0, 1487.67017741838)
[10:59:49] [Server thread/INFO] [net.minecraft.server.dedicated.DedicatedServer]: Owner bonnydust joined the game
[11:08:48] [Server thread/INFO] [minecolonies]: WorkOrderBuild at location BlockPos{x=285, y=80, z=1459} is using schematics/Medieval DarkOak/Citizen1 instead of schematics/Medieval DarkOak/Citizen1
[11:10:41] [Server thread/INFO] [minecolonies]: [BarbarianEvent]: Spawning: 1133 -720
[11:10:41] [Server thread/WARN] [minecolonies]: Spawning barbarian at: BlockPos{x=1133, y=63, z=-720}
[11:10:41] [Server thread/WARN] [minecolonies]: Spawning barbarian at: BlockPos{x=1133, y=63, z=-720}
[11:12:12] [Server thread/WARN] [minecolonies]: Spawning barbarian at: BlockPos{x=1183, y=62, z=-713}
[11:12:12] [Server thread/WARN] [minecolonies]: Spawning barbarian at: BlockPos{x=1187, y=62, z=-711}
[11:12:31] [Server thread/WARN] [minecolonies]: Spawning barbarian at: BlockPos{x=1197, y=62, z=-711}
[11:12:31] [Server thread/WARN] [minecolonies]: Spawning barbarian at: BlockPos{x=1190, y=62, z=-713}
[11:13:16] [Server thread/WARN] [minecolonies]: Spawning barbarian at: BlockPos{x=1211, y=62, z=-713}
[11:13:18] [Server thread/INFO] [minecolonies]: Teleported stuck citizen Ruben Z. Setherby from colony: 3 to target location
[11:15:18] [Server thread/INFO] [minecolonies]: Teleported stuck citizen Timothy I. Lovegrove from colony: 3 to target location
[11:15:36] [Server thread/INFO] [minecolonies]: Teleported stuck citizen Ruben Z. Setherby from colony: 3 to target location
[11:17:37] [Server thread/INFO] [minecolonies]: Teleported stuck citizen Timothy I. Lovegrove from colony: 3 to target location
[11:17:50] [Server thread/INFO] [minecolonies]: Teleported stuck citizen Ruben Z. Setherby from colony: 3 to target location
[11:20:14] [Server thread/INFO] [gravestone-extended]: Generate grave at 919x-729
[11:20:56] [Server thread/INFO] [net.minecraft.server.dedicated.DedicatedServer]: [bonnydust: Set own game mode to Spectator Mode]
[11:22:24] [Server thread/INFO] [net.minecraft.server.dedicated.DedicatedServer]: [bonnydust: Set own game mode to Creative Mode]
[11:22:48] [Server thread/INFO] [minecolonies]: Teleported stuck citizen Ruben Z. Setherby from colony: 3 to target location
[11:22:50] [Server thread/INFO] [minecolonies]: Teleported stuck citizen Timothy I. Lovegrove from colony: 3 to target location
[11:25:06] [Server thread/INFO] [minecolonies]: Teleported stuck citizen Ruben Z. Setherby from colony: 3 to target location
[11:25:17] [Server thread/INFO] [minecolonies]: Teleported stuck citizen Timothy I. Lovegrove from colony: 3 to target location
[11:27:22] [Server thread/INFO] [minecolonies]: Teleported stuck citizen Ruben Z. Setherby from colony: 3 to target location
[11:27:34] [Server thread/INFO] [minecolonies]: Teleported stuck citizen Timothy I. Lovegrove from colony: 3 to target location
[11:28:02] [Server thread/INFO] [net.minecraft.server.dedicated.DedicatedServer]: [bonnydust: Set own game mode to Spectator Mode]
[11:28:16] [Server thread/INFO] [net.minecraft.server.dedicated.DedicatedServer]: [bonnydust: Set own game mode to Creative Mode]
[11:28:38] [Server thread/INFO] [net.minecraft.server.dedicated.DedicatedServer]: [bonnydust: Set own game mode to Spectator Mode]
[11:28:44] [Server thread/INFO] [net.minecraft.server.dedicated.DedicatedServer]: [bonnydust: Set own game mode to Creative Mode]
[11:29:30] [Server thread/INFO] [minecolonies]: Teleported stuck citizen Ruben Z. Setherby from colony: 3 to target location
[11:30:04] [Server thread/INFO] [minecolonies]: Teleported stuck citizen Timothy I. Lovegrove from colony: 3 to target location
[11:31:32] [Server thread/INFO] [net.minecraft.server.dedicated.DedicatedServer]: [bonnydust: Set own game mode to Spectator Mode]
[11:31:47] [Server thread/INFO] [net.minecraft.server.dedicated.DedicatedServer]: [bonnydust: Set own game mode to Creative Mode]
[11:31:56] [Server thread/INFO] [minecolonies]: Teleported stuck citizen Ruben Z. Setherby from colony: 3 to target location
[11:32:20] [Server thread/INFO] [minecolonies]: Teleported stuck citizen Timothy I. Lovegrove from colony: 3 to target location
[11:34:18] [Server thread/INFO] [minecolonies]: Teleported stuck citizen Ruben Z. Setherby from colony: 3 to target location
[11:34:39] [Server thread/INFO] [minecolonies]: Teleported stuck citizen Timothy I. Lovegrove from colony: 3 to target location
[11:35:30] [Server thread/INFO] [net.minecraft.server.dedicated.DedicatedServer]: [bonnydust: Set own game mode to Spectator Mode]
[11:35:36] [Server thread/INFO] [net.minecraft.server.dedicated.DedicatedServer]: [bonnydust: Set own game mode to Creative Mode]
[11:36:14] [Server thread/INFO] [net.minecraft.server.dedicated.DedicatedServer]: [bonnydust: Set own game mode to Spectator Mode]
[11:36:21] [Server thread/INFO] [net.minecraft.server.dedicated.DedicatedServer]: [bonnydust: Set own game mode to Creative Mode]
[11:36:33] [Server thread/INFO] [minecolonies]: Teleported stuck citizen Ruben Z. Setherby from colony: 3 to target location
[11:36:55] [Server thread/INFO] [minecolonies]: Teleported stuck citizen Timothy I. Lovegrove from colony: 3 to target location
[11:37:01] [Server thread/INFO] [net.minecraft.server.dedicated.DedicatedServer]: [bonnydust: Set own game mode to Survival Mode]
[11:38:14] [Server thread/INFO] [net.minecraft.server.dedicated.DedicatedServer]: [bonnydust: Set own game mode to Spectator Mode]
[11:38:22] [Server thread/INFO] [net.minecraft.server.dedicated.DedicatedServer]: [bonnydust: Set own game mode to Creative Mode]
[11:38:46] [Server thread/INFO] [minecolonies]: Teleported stuck citizen Ruben Z. Setherby from colony: 3 to target location
[11:39:10] [Server thread/INFO] [minecolonies]: Teleported stuck citizen Timothy I. Lovegrove from colony: 3 to target location
[11:39:26] [Server thread/INFO] [minecolonies]: Spawning at colony center: 320 1476
[11:39:26] [Server thread/INFO] [minecolonies]: [BarbarianEvent]: Spawning: 320 1476
[11:39:34] [Server thread/INFO] [net.minecraft.server.dedicated.DedicatedServer]: [bonnydust: Set own game mode to Spectator Mode]
[11:39:43] [Server thread/INFO] [net.minecraft.server.dedicated.DedicatedServer]: [bonnydust: Set own game mode to Survival Mode]
[11:41:00] [Server thread/INFO] [minecolonies]: Teleported stuck citizen Ruben Z. Setherby from colony: 3 to target location
[11:41:33] [Server thread/INFO] [minecolonies]: Teleported stuck citizen Timothy I. Lovegrove from colony: 3 to target location
[11:43:02] [Server thread/INFO] [net.minecraft.server.dedicated.DedicatedServer]: [bonnydust: Set own game mode to Spectator Mode]
[11:43:12] [Server thread/INFO] [net.minecraft.server.dedicated.DedicatedServer]: [bonnydust: Set own game mode to Creative Mode]
[11:43:29] [Server thread/INFO] [minecolonies]: Teleported stuck citizen Ruben Z. Setherby from colony: 3 to target location
[11:43:56] [Server thread/INFO] [minecolonies]: Teleported stuck citizen Timothy I. Lovegrove from colony: 3 to target location
[11:45:44] [Server thread/INFO] [minecolonies]: Teleported stuck citizen Ruben Z. Setherby from colony: 3 to target location
[11:46:16] [Server thread/INFO] [minecolonies]: Teleported stuck citizen Timothy I. Lovegrove from colony: 3 to target location
[11:47:12] [Server thread/INFO] [net.minecraft.server.dedicated.DedicatedServer]: [bonnydust: Set own game mode to Spectator Mode]
[11:47:29] [Server thread/INFO] [net.minecraft.server.dedicated.DedicatedServer]: [bonnydust: Set own game mode to Creative Mode]
[11:48:02] [Server thread/INFO] [minecolonies]: Teleported stuck citizen Ruben Z. Setherby from colony: 3 to target location
[11:48:27] [Server thread/INFO] [minecolonies]: Teleported stuck citizen Timothy I. Lovegrove from colony: 3 to target location
[11:48:44] [Server thread/INFO] [net.minecraft.server.dedicated.DedicatedServer]: [bonnydust: Set own game mode to Survival Mode]
[11:50:15] [Server thread/INFO] [minecolonies]: Teleported stuck citizen Ruben Z. Setherby from colony: 3 to target location
[11:50:20] [Server thread/INFO] [minecolonies]: [BarbarianEvent]: Spawning: 1359 -721
[11:50:20] [Server thread/WARN] [minecolonies]: Spawning barbarian at: BlockPos{x=1359, y=63, z=-721}
[11:50:20] [Server thread/WARN] [minecolonies]: Spawning barbarian at: BlockPos{x=1359, y=63, z=-721}
[11:50:42] [Server thread/INFO] [minecolonies]: Teleported stuck citizen Timothy I. Lovegrove from colony: 3 to target location
[11:52:32] [Server thread/INFO] [minecolonies]: Teleported stuck citizen Ruben Z. Setherby from colony: 3 to target location
[11:52:58] [Server thread/INFO] [minecolonies]: Teleported stuck citizen Timothy I. Lovegrove from colony: 3 to target location
[11:54:46] [Server thread/INFO] [minecolonies]: Teleported stuck citizen Ruben Z. Setherby from colony: 3 to target location
[11:55:14] [Server thread/INFO] [minecolonies]: Teleported stuck citizen Timothy I. Lovegrove from colony: 3 to target location
[11:56:58] [Server thread/INFO] [minecolonies]: Teleported stuck citizen Ruben Z. Setherby from colony: 3 to target location
[11:57:29] [Server thread/INFO] [minecolonies]: Teleported stuck citizen Timothy I. Lovegrove from colony: 3 to target location
[11:59:14] [Server thread/INFO] [minecolonies]: Teleported stuck citizen Ruben Z. Setherby from colony: 3 to target location
[11:59:51] [Server thread/INFO] [minecolonies]: Teleported stuck citizen Timothy I. Lovegrove from colony: 3 to target location
[12:00:43] [Server thread/INFO] [net.minecraft.network.NetHandlerPlayServer]: bonnydust lost connection: Disconnected
[12:00:43] [Server thread/INFO] [net.minecraft.server.dedicated.DedicatedServer]: Owner bonnydust left the game
[12:00:53] [Server thread/INFO] [net.minecraft.server.dedicated.DedicatedServer]: Stopping the server
[12:00:53] [Server thread/INFO] [net.minecraft.server.MinecraftServer]: Stopping server
[12:00:53] [Server thread/INFO] [net.minecraft.server.MinecraftServer]: Saving players
[12:00:53] [Server thread/INFO] [net.minecraft.server.MinecraftServer]: Saving worlds
[12:00:53] [Server thread/INFO] [net.minecraft.server.MinecraftServer]: Saving chunks for level 'A New World'/overworld
[12:00:54] [Server thread/INFO] [FML]: Unloading dimension 0
```

### Steps to reproduce the problem

1. Run raid-tonight command for the colony or allow a raid to occur naturally.
2. Kill barbarians.
3. If necessary repeat until you get the issue.
4. Additionally sometimes the countdown does match actual barbarian count but all or some of the barbarians are invisible.

![2018-08-12_11 33 48](https://user-images.githubusercontent.com/41415176/43997901-6af3b55a-9e2c-11e8-9a15-d85db1cfa75d.png)
![2018-08-12_11 34 37](https://user-images.githubusercontent.com/41415176/43997902-6b2f62a8-9e2c-11e8-9501-798bfa16f6bf.png)
![2018-08-12_11 34 53](https://user-images.githubusercontent.com/41415176/43997903-6b6876d8-9e2c-11e8-821c-eccee94fb2d5.png)
![2018-08-12_11 51 07](https://user-images.githubusercontent.com/41415176/43997904-6ba320c6-9e2c-11e8-9551-dfd268941340.png)
![2018-08-12_11 55 34](https://user-images.githubusercontent.com/41415176/43997905-6bdef2cc-9e2c-11e8-92b5-43034cefd4d9.png)

EDIT: Nightenom - code block

**Comment by Raycoms** [comment]
Can you check if the newest update fixes it?

2018-08-11 23:45 GMT-03:00 QueenBonnydust <notifications@github.com>:

**Comment by QueenBonnydust** [comment]
So far the same issue does not appear, however it has been replaced with an issue where the barbarians are marked as killed and colony successfully defended before anyone (or any guards) have even touched them, in one case twice over. See screenshots. After taking a screen shot to show that barbarians still existed in the colony and had indeed not been killed, I killed them and received correct countdown and colony successfully defended message.

Should I post this as a new issue?

![2018-08-12_17 02 39](https://user-images.githubusercontent.com/41415176/43999613-7fd32b00-9e53-11e8-9c07-784da8f2136a.png)
![2018-08-12_17 04 01](https://user-images.githubusercontent.com/41415176/43999614-800e456e-9e53-11e8-9af3-53767d6880cc.png)
![2018-08-12_17 13 33](https://user-images.githubusercontent.com/41415176/43999615-80462376-9e53-11e8-9c9d-b48c0bc360bc.png)
![2018-08-12_17 14 05](https://user-images.githubusercontent.com/41415176/43999616-80809d9e-9e53-11e8-8190-a76fc5cb0ec4.png)
